rflx/pyrflx/typevalue.py

```diff
@@ -601,22 +607,47 @@ def __eq__(self, other: object) -> bool:
     def equal_type(self, other: Type) -> bool:
         return self.identifier == other.identifier
 
+    @property
+    def covered_links(self) -> List[Link]:
+        return self._covered_links
+
+    def sdu_messages(self) -> List["MessageValue"]:
+        sdu_messages: List["MessageValue"] = []
+        for field in self._fields.values():
+            typeval = field.typeval
+            if isinstance(typeval, ArrayValue) and isinstance(typeval.element_type, Message):
+                sdu_messages.extend(cast(List[MessageValue], typeval.value))
+            if isinstance(typeval, OpaqueValue) and typeval.nested_message is not None:
+                sdu_messages.append(typeval.nested_message)
+        return sdu_messages
+
     def _valid_refinement_condition(self, refinement: "RefinementValue") -> bool:
         return self.__simplified(refinement.condition) == TRUE
 
-    def _next_field(self, fld: str) -> str:
-        if fld == FINAL.name:
-            return ""
-        if self._skip_verification and self._fields[fld].next:
-            return self._fields[fld].next
-        if fld == INITIAL.name:
+    def _next_link(self, source_field_name: str) -> Optional[Link]:
+        field = Field(source_field_name)
+        if field == INITIAL:
             links = self._type.outgoing(INITIAL)
-            return links[0].target.name if links else FINAL.name
+            return links[0] if links else None
+        if field == FINAL:
+            return None
+
+        next_link = None
+        for link in self._type.outgoing(field):
+            if self.__simplified(link.condition) == TRUE:
+                next_link = link
+                break
+        return next_link
 
-        for l in self._type.outgoing(Field(fld)):
-            if self.__simplified(l.condition) == TRUE:
-                return l.target.name
-        return ""
+    def _next_field(self, field_name: str) -> Tuple[str, Optional[Link]]:
+        if self._skip_verification and field_name != FINAL.name and self._fields[field_name].next:
+            return self._fields[field_name].next, None
+
+        next_link = self._next_link(field_name)
+        if next_link is None and field_name == INITIAL.name:
+            # the INITIAL field has no outgoing links in case of a NULL message
+            return FINAL.name, None
+        return next_link.target.name if next_link is not None else "", next_link
```

Comment:
`_next_link` is only used once, namly in `_next_field`. What is the intention of moving some code into this separate function? I don't understand how splitting this function "avoid[s] introducing a boolean flag coverage to the `next_field` method".

Reply:
Oh, it does not. Sorry ;) I made a mistake while rebasing and dropped a commit.
I'll fix that. The functionality of the `_next_link` method is moved back into `next_field` but the  general behaviour is the same (that means the return of  Tuple[str, Optional[Link]])